nodes/field/scalar_field_curvature.py

```diff
@@ -0,0 +1,74 @@
+
+import numpy as np
+
+import bpy
+from bpy.props import FloatProperty, EnumProperty, BoolProperty, IntProperty, StringProperty
+
+from sverchok.node_tree import SverchCustomTreeNode, throttled
+from sverchok.data_structure import updateNode, zip_long_repeat, match_long_repeat, ensure_nesting_level
+from sverchok.utils.logging import info, exception
+
+from sverchok.utils.field.scalar import SvScalarField, SvScalarFieldGaussCurvature, SvScalarFieldMeanCurvature, SvScalarFieldPrincipalCurvature, ScalarFieldCurvatureCalculator
+
+class SvScalarFieldCurvatureNode(bpy.types.Node, SverchCustomTreeNode):
+    """
+    Triggers: Scalar Field Curvature
+    Tooltip: Scalar Field Curvature
+    """
+    bl_idname = 'SvScalarFieldCurvatureNode'
+    bl_label = 'Scalar Field Curvature'
+    bl_icon = 'OUTLINER_OB_EMPTY'
+    sv_icon = 'SV_SCALAR_FIELD_MATH'
+
+    def sv_init(self, context):
+        self.inputs.new('SvScalarFieldSocket', "Field")
+        self.outputs.new('SvScalarFieldSocket', "Gauss")
+        self.outputs.new('SvScalarFieldSocket', "Mean")
+        self.outputs.new('SvScalarFieldSocket', "Curvature1")
+        self.outputs.new('SvScalarFieldSocket', "Curvature2")
+
+    step : FloatProperty(
+            name = "Step",
+            default = 0.001,
+            precision = 4,
+            update = updateNode)
+
+    def draw_buttons(self, context, layout):
+        layout.prop(self, 'step')
+
+    def process(self):
+        if not any(socket.is_linked for socket in self.outputs):
+            return
```

Comment:
Let to the update system to decide, whether it is worth call update or not.

Reply:
Currently most of nodes do similar check. I agree this is better for update system to decide, but as far as I understand in it's current state it is not clever enough to do not recalculate nodes that do not have their outputs connected. Moreover, there are nodes that should do some work even when their outputs are not connected - all kinds of Viewers.
When you'll merge more clever update system, I think we'll all be happy to remove all these checks from all nodes :)

Reply:
You understand it wrong. I'm saying about how exactly current update system works. It is redundant code relatively to current update system.

Reply:
Also some nodes calculate different things based on the outputs connected. For example the Box node does not compute the polygon data if the polygons output is not connected, so I'm not sure how that could be handled from the update system...

Reply:
In my current attempt I have sacrificed such optimisations. Because for solving this issue workflow of all nodes should be reconsidered.